### PR TITLE
refactor(table): change table component to use generics

### LINKED ISSUE #549: Table component should use generics
## 🐛 Bug Report
Currently, the table component does not properly use generics. It creates a type `T` that makes it seem like it accepts generics, but it does not. 

This issue is to make the table understand generics. Everywhere type `T` is, we should replace it with a generic. 

See `Select.tsx` for examples of generics.

### PR DESCRIPTION
**Pull request by rsousaj**
Fixes #549 .

**Changes proposed in this pull request:**
- Table components now using Generics

### CODE REVIEW

**Comment by jackcmeyer** [comment]
change looks good to me. Just waiting on @rsousaj to sign the CLA.

**Comment by rsousaj** [comment]
Sorry about that, I forgot that frontend and components are different projects.

But now that I've already signed it, it still look pending. Am I messing around ?

File: src/components/Table/Table.tsx [hunk 1]
```diff
@@ -3,9 +3,11 @@ import { ButtonVariant } from 'src/interfaces'
 
 import { Button } from '../Button'
 
-type T = { [key: string]: any }
+interface Row {
+  [key: string]: any
```
Comment (fox1t) [inline src/components/Table/Table.tsx:7]:
```suggestion
  [key: string]: unknown
```

**Review by fox1t — CHANGES_REQUESTED** [review]

**Comment by hugo1707** [comment]
Hello, 
I suggest changing it to ReactNode to fix these errors

**Comment by morrme** [comment]
@hugo1707 @rsousaj  What is the status of this PR?

**Comment by rsousaj** [comment]
> @hugo1707 @rsousaj What is the status of this PR?

Hi @morrme, what do you think about @hugo1707 question ? Should we change it to ReactNode instead of 'unknown' ?

**Review by blestab — DISMISSED** [review]

**Comment by tehKapa** [comment]
> > @hugo1707 @rsousaj What is the status of this PR?
> 
> Hi @morrme, what do you think about @hugo1707 question ? Should we change it to ReactNode instead of 'unknown' ?

@rsousaj try it. in case it should continue to give error

**Comment by rsousaj** [comment]
> > > @hugo1707 @rsousaj What is the status of this PR?
> > 
> > 
> > Hi @morrme, what do you think about @hugo1707 question ? Should we change it to ReactNode instead of 'unknown' ?
> 
> @rsousaj try it. in case it should continue to give error

Hi guys, sorry for taking too long to reply. I didn't have internet last couple of days. We'll try that.

**Review by tehKapa — APPROVED** [review]

**Pull request merged**